### PR TITLE
Don't allow Money to represent -0.00

### PR DESCRIPTION
**Pull request by nick-mcdonald**
Prevent storing negative zero in a Money object.

@Shopify/merchant-data please review

### CODE REVIEW

**Comment by stephenminded** [comment]
:+1:

**Comment by jamesemoody** [comment]
Awesome. :boat:

**Comment by karanthukral** [comment]
:ship: :shipit:

File: lib/money/money.rb [hunk 1]
```diff
@@ -10,6 +10,7 @@ def initialize(value = 0)
     raise ArgumentError if value.respond_to?(:nan?) && value.nan?
 
     @value = value_to_decimal(value).round(2)
+    @value = 0.to_d if @value.sign == BigDecimal::SIGN_NEGATIVE_ZERO
```
Comment (jeromecornet) [inline lib/money/money.rb]: Whaaat ? mind blown

**Comment by jeromecornet** [comment]
:+1:

**Pull request merged**